### PR TITLE
Update SB CI to alpine 3.20

### PR DESCRIPTION
**Pull request by MichaelSimons**
Related to https://github.com/dotnet/source-build/issues/4399

Updated the Alpine_Online_MsftSdk leg to use Alpine 3.20 (latest).  I kept the Alpine_Offline_PreviousSourceBuiltSdk let using Alpine 3.19 (latest - 1).  This was done for two reasons.

1. It provides more coverage - multiple alpine versions
2. It supports frequent updates to the alpine versions as it provides us a way to retrieve the artifacts to build the PreviousSourceBuiltSdk with.  You can retrieve them from the latest leg.

### CODE REVIEW

**Comment by MichaelSimons** [comment]
This is ready for review now.

**Review by ellahathaway — APPROVED** [review]

File: eng/pipelines/templates/variables/vmr-build.yml [hunk 1]
```diff
@@ -14,7 +14,9 @@ variables:
 
 - name: almaLinuxContainer
   value: mcr.microsoft.com/dotnet-buildtools/prereqs:almalinux-8-source-build
-- name: alpineContainer
+- name: alpineLatestContainer
+  value: mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.20-withnode
+- name: alpineLatestMinusOneContainer
```
Comment (mthalman) [inline eng/pipelines/templates/variables/vmr-build.yml:19]: I'd prefer a name like `alpinePrevious` instead of `alpineLatestMinusOne`.

Comment (MichaelSimons) [reply eng/pipelines/templates/variables/vmr-build.yml:19]: I struggled with the naming.  I like your suggestion better.  Will update.

Comment (MichaelSimons) [reply eng/pipelines/templates/variables/vmr-build.yml]: Fixed in https://github.com/dotnet/sdk/pull/42119/commits/201986f511b3232b4e331d4ac41d81a894532715

**Review by mthalman — APPROVED** [review]

**Pull request merged**